universal-application-tool-0.0.1/app/views/questiontypes/RepeaterQuestionRenderer.java

```diff
@@ -36,43 +37,70 @@ public RepeaterQuestionRenderer(ApplicantQuestion question) {
 
   @Override
   public Tag render(Messages messages) {
-    ContainerTag repeaterFields = div().withId(REPEATER_FIELDS_ID);
-    // TODO: add each answer as a repeaterField
-    repeaterFields.with(repeaterField(PLACEHOLDER, Optional.empty(), 0));
+    RepeaterQuestion enumeratorQuestion = question.createEnumeratorQuestion();
+    ImmutableList<String> entityNames = enumeratorQuestion.getEntityNames();
+
+    ContainerTag enumeratorFields = div().withId(ENUMERATOR_FIELDS_ID);
+    for (int index = 0; index < entityNames.size(); index++) {
+      enumeratorFields.with(existingEnumeratorField(Optional.of(entityNames.get(index)), index));
+    }
 
     return div()
         .withClasses(Styles.MX_AUTO, Styles.W_MAX)
         .with(
             div()
                 .withClasses(ReferenceClasses.APPLICANT_QUESTION_TEXT)
                 .withText(question.getQuestionText()),
+            div()
+                .withId(PLACEHOLDER_ID)
+                .withClass(Styles.HIDDEN)
+                .withText(enumeratorQuestion.getPlaceholder()),
             div()
                 .withClasses(
                     ReferenceClasses.APPLICANT_QUESTION_HELP_TEXT,
                     Styles.TEXT_BASE,
                     Styles.FONT_THIN,
                     Styles.MB_2)
                 .withText(question.getQuestionHelpText()),
-            repeaterFields,
-            button(ADD_ELEMENT_BUTTON_ID, messages.at("button.addRepeaterEntity")))
-        .withType("button");
+            enumeratorFields,
+            button(ADD_ELEMENT_BUTTON_ID, messages.at("button.addEnumeratorEntity"))
+                .withType("button"),
```

Comment:
I don't think `withType` is necessary here since [the underlying `button` method](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/views/BaseHtmlView.java#L50) sets the type.

I'm curious if this `withType` call was here for some other purpose though, since it looks like it was previously modifying a div rather than this button, based on the parens...

Reply:
fixed!

Really not sure where this came from 😦